src/targets/gpu/lowering.cpp

```diff
@@ -376,6 +377,26 @@ struct miopen_apply
             return mod->replace_instruction(ins, ins->get_operator(), inputs, ins->module_inputs());
         });
     }
+
+    /**
+     *  Adds reshape lazy to reshape ops that can be aliased instead of copied
+     */
+    void add_reshape_lazy_op()
+    {
+        apply_map.emplace("reshape", [=](instruction_ref ins) {
+            auto lazy_ins = mod->replace_instruction(
+                ins,
+                make_op("reshape_lazy", {{"dims", {ins->get_operator().to_value().at("dims")}}}),
+                ins->inputs(),
+                ins->module_inputs());
```

Comment:
A contiguous should be inserted before the reshape_lazy as well. There should be a contiguous inserted before and after the reshape_lazy.